reassembler.go

```diff
@@ -140,35 +141,25 @@ func (r *Reassembler) callback(events []*event, lost int) {
 	}
 }
 
-type sequenceNum uint32
-
-// Type - sequenceNumSlice
-
-// maxSortRange defines the maximum range that sequence number can differ
-// before being considered to have rolled over. When two values differ by more
-// than this constant, the larger values is treated as being less.
 const maxSortRange = 1<<24 - 1
 
-type sequenceNumSlice []sequenceNum
+type heapInt []int
 
-func (p sequenceNumSlice) Len() int      { return len(p) }
-func (p sequenceNumSlice) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
-func (p sequenceNumSlice) Sort()         { sort.Sort(p) }
-
-func (p sequenceNumSlice) Less(i, j int) bool {
-	// Handle sequence number rollover.
-	diff := abs(int64(p[i]) - int64(p[j]))
-	if diff > maxSortRange {
-		return p[i] > p[j]
+func (h heapInt) Len() int { return len(h) }
+func (h heapInt) Less(i, j int) bool {
+	if math.Abs(float64(h[i]-h[j])) > maxSortRange {
+		return h[i] > h[j]
 	}
-
-	return p[i] < p[j]
+	return h[i] < h[j]
 }
-
-func abs(x int64) int64 {
-	if x < 0 {
-		return -x
-	}
+func (h heapInt) Swap(i, j int)       { h[i], h[j] = h[j], h[i] }
+func (h *heapInt) Push(x interface{}) { *h = append(*h, x.(int)) }
+
+func (h *heapInt) Pop() interface{} {
```

Comment:
if we have heapInt, returning int instead of interface{} will be more logical

Reply:
We can't change type. It's implementation of the interface